shower.js

```diff
@@ -862,40 +910,16 @@ window.shower = (function(window, document, undefined) {
 	}
 
 	// Event handlers
+	window.addEventListener('popstate', function() {
+		shower._setTitle();
+	}, false);
+
 	window.addEventListener('DOMContentLoaded', function() {
 		shower.
 			init().
 			run();
 	}, false);
 
-	window.addEventListener('popstate', function() {
-		var currentSlideNumber = shower.getCurrentSlideNumber(),
-			isSlideMode = document.body.classList.contains('full') || shower.isSlideMode();
-
-		if (shower._getSlideTitle(currentSlideNumber)) {
-			document.title = shower._getSlideTitle(currentSlideNumber) + ' — ' + presentationTitle;
-		} else {
-			document.title = presentationTitle;
-		}
-
-		// Go to first slide, if hash id is invalid or isn't set.
-		// Same check is located in DOMContentLoaded event,
-		// but it not fires on hash change
-		if (isSlideMode && currentSlideNumber === -1) {
-			shower.go(0);
-
-		// In List mode, go to first slide only if hash id is invalid.
-		} else if (currentSlideNumber === -1 && url.hash !== '') {
-			shower.go(0);
-		}
```

Comment:
I guess it’s something that couldn’t be fixed since it’s not possible to use History API anymore. Am I right?